src/templates/ProductPage.js

```diff
@@ -1,16 +1,41 @@
-import React from 'react'
+import React, { useContext, useEffect, useState } from 'react'
 import { graphql, withPrefix } from 'gatsby'
 
 import SEO from '../components/SEO'
 import Photo from '../components/Photo'
 import Badge from '../components/Badge'
 import AddToCart from '../components/AddToCart'
+import { Shopkit } from '../shopkit'
 
 function ProductPage({ data: { product } }) {
+  const { moltin } = useContext(Shopkit)
+  const [inventoryAvailability, setInventoryAvailability] = useState(0)
+  const [inventoryLoading, setInventoryLoading] = useState(true)
+  const [inventoryError, setInventoryError] = useState(false)
+
+  useEffect(() => {
+    async function getProductInventory() {
+      try {
+        const {
+          data: { available }
+        } = await moltin.get(`inventories/${product.id}`)
+
+        setInventoryLoading(false)
+        setInventoryAvailability(available)
+      } catch (error) {
+        setInventoryError(error)
```

Comment:
I'm unsure how this will display the error further down. As `errors` will be an array.

Reply:
Also I get an `uncaught Promise` error when trying to add more items than I have in stock.

Reply:
> Also I get an `uncaught Promise` error when trying to add more items than I have in stock.

I don't see this. I see a `201`, which is probably even more concerning.